pkg/controller/garbagecollector/graph.go

```diff
@@ -55,10 +55,15 @@ type node struct {
 	owners []metav1.OwnerReference
 }
 
-func (n *node) setBeingDeleted(v bool) {
+// WARNING: the following setters/getters use different locks, so the return
+// values of the getters can be inconsistent.
```

Comment:
This is appropriately scary. Can we add it to the comment on the `node` struct instead of having it loose here?